doc/source/whatsnew/v3.0.0.rst

```diff
@@ -1122,6 +1122,7 @@ Other
 - Bug in printing a :class:`DataFrame` with a :class:`DataFrame` stored in :attr:`DataFrame.attrs` raised a ``ValueError`` (:issue:`60455`)
 - Bug in printing a :class:`Series` with a :class:`DataFrame` stored in :attr:`Series.attrs` raised a ``ValueError`` (:issue:`60568`)
 - Fixed bug where the :class:`DataFrame` constructor misclassified array-like objects with a ``.name`` attribute as :class:`Series` or :class:`Index` (:issue:`61443`)
+- Fixed bug where the :meth:`Series.rank` doesn't rank the values with object dtype and small values properly (:issue:`62036`)
```

Comment:
Fixed bug in :meth:`Series.rank` with ``object`` dtype and extremely small ``float`` values